src/client/Client.cc

```diff
@@ -8930,14 +8949,17 @@ loff_t Client::_lseek(Fh *f, loff_t offset, int whence)
     pos = in->size + offset;
     break;
 
+#ifdef SEEK_DATA
   case SEEK_DATA:
     if (offset < 0 || static_cast<uint64_t>(offset) >= in->size) {
       r = -ENXIO;
```

Comment:
Patch itself looks fine, but the existing code looks wrong. Shouldn't this just return -ENXIO?

Reply:
Opened this to track the bad error handling there: https://tracker.ceph.com/issues/44021